ntoskrnl/ke/bug.c

```diff
@@ -334,6 +334,420 @@ KiInitializeBugCheck(VOID)
     }
 }
 
+PCHAR
+NTAPI
+KeGetBugMessageName(IN ULONG MessageId)
+{
+    /* Generated with https://github.com/feel-the-dz3n/kebughtoc */
```

Comment:
I've just discovered that the `-x` [option of the message compiler](https://learn.microsoft.com/en-us/windows/win32/wes/message-compiler--mc-exe-#arguments-specific-to-message-text-files) (works for both MC.EXE and GNU windres.exe), allows to generate a separate source file containing a C-style mapping array of values in a manner similar to what you want to achieve:
```c
struct testbugcodesSymbolicName
{
  ULONG MessageId;
  char *SymbolicName;
} testbugcodesSymbolicNames[] =
{
  {(ULONG) WINDOWS_NT_BANNER, "WINDOWS_NT_BANNER" },
  {(ULONG) WINDOWS_NT_CSD_STRING, "WINDOWS_NT_CSD_STRING" },
....
  { (ULONG) 0xffffffff, NULL }
};
```
To find the string, one would have to write a for/while loop, instead of a switch.